libraries/AP_FETtecOneWire/AP_FETtecOneWire.cpp

```diff
@@ -100,32 +100,35 @@ void AP_FETtecOneWire::update()
         return;
     }
 
-    if (!_uart->is_dma_enabled()) {
-        GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "FTW UART needs DMA");
+    if (_update_loop_decimator-- == 0) {
+        if (!_uart->is_dma_enabled()) {
+            GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "FTW UART needs DMA");
+        }
+        // tell SRV_Channels about ESC capabilities
+        _mask = uint16_t(motor_mask.get());
+        SRV_Channels::set_digital_outputs(_mask, 0);
+
+        _update_loop_decimator = 800; // once every 2 seconds
     }
-    // tell SRV_Channels about ESC capabilities
-    const uint16_t mask = uint16_t(motor_mask.get());
-    SRV_Channels::set_digital_outputs(mask, 0);
 
     // get ESC set points, stop as soon as there is a gap
-    uint16_t motor_pwm[MOTOR_COUNT_MAX] = {1000};
+    uint16_t motor_pwm[MOTOR_COUNT_MAX];
+    uint16_t smask = _mask;
+    uint8_t nr_escs = 0;
 
     for (uint8_t i = 0; i < MOTOR_COUNT_MAX; i++) {
         SRV_Channel* c = SRV_Channels::srv_channel(i);
-        if (c == nullptr) {
+        if (c == nullptr || (smask & 0x01) == 0x00) {
             break;
         }
-        motor_pwm[i] = constrain_int16(c->get_output_pwm(), 0, 2000);
+        motor_pwm[i] = constrain_int16(c->get_output_pwm(), 1000, 2000);

```

Comment:
is there no 3D mode in Ardupilot?

Reply:
If I understood @tridge correctly there is no unified concept for reversion. There are multiple concetpts for reversion, each ESC implements its own.
This might change in the future, but currently it's the wild-west.

Reply:
I have implemented it now. We can reverse the rotation direction per parameter now.